File: lib/domain/entities/recommendation.dart
import 'practice_module.dart';

class Competency {
  const Competency({
    required this.moduleId,
    required this.moduleTitle,
    required this.mastery,
    required this.attempts,
  });

  final int moduleId;
  final String moduleTitle;
  final double mastery;
  final int attempts;
}

class Recommendation {
  const Recommendation({
    required this.studentId,
    required this.riskLevel,
    required this.averageScore,
    required this.attemptCount,
    required this.reasons,
    required this.competencies,
    this.nextModule,
    required this.reason,
    this.aiSuggestion,
  });

  final int studentId;
  final String riskLevel;
  final double averageScore;
  final int attemptCount;
  final List<String> reasons;
  final List<Competency> competencies;
  final PracticeModule? nextModule;
  final String reason;
  final String? aiSuggestion;
}

File: vocalearn-backend/app/schemas/recommendation.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel

from app.schemas.module import ModuleOut


class CompetencyOut(BaseModel):
    module_id: int
    module_title: str
    course_id: int
    mastery: float  # 0..1
    attempts: int


class RiskOut(BaseModel):
    level: str  # low | medium | high
    average_score: float
    attempt_count: int
    reasons: list[str]


class RecommendationOut(BaseModel):
    student_id: int
    generated_at: datetime
    risk: RiskOut
    competencies: list[CompetencyOut]
    next_module: Optional[ModuleOut] = None
    reason: str
    ai_suggestion: Optional[str] = None
